OQ_Toolkit/OQ_ARVRController/scripts/Feature_RigidBodyGrab.gd

```diff
@@ -1,65 +1,56 @@
 extends Spatial
 
 var controller : ARVRController = null;
-
 var grab_area : Area = null;
 var held_object = null;
 var held_object_data = {};
-
 var grab_mesh : MeshInstance = null;
-export var reparent_mesh = false;
-
-onready var grabbed_object_script = preload("helper_grabbed_RigidBody.gd");
-
 
 enum {
 	GRABTYPE_VELOCITY,
 	GRABTYPE_PINJOINT, #!!TODO: not yet working; I first need to figure out how joints work
 }
-
 var grab_type = GRABTYPE_VELOCITY;
 
+onready var main_node = get_node("/root/Main")
+
+export var reparent_mesh = false;
+
 func _ready():
 	controller = get_parent();
 	if (not controller is ARVRController):
 		vr.log_error(" in Feature_RigidBodyGrab: parent not ARVRController.");
 	grab_area = $GrabArea;
+	
+	controller.connect("button_pressed", self, "_on_ARVRController_button_pressed")
```

Comment:
At the moment we can't use signals for button events for core features as this is not yet supported by the `Feature_VRSimulator.tscn` and also not the playback of `Feature_VRRecorder.tscn`. So it needs to still be checked as previously with `if (controller._button_just_pressed(vr.CONTROLLER_BUTTON.GRIP_TRIGGER)):` in the process

Reply:
I see. As you know, it would be much better practice to go with the signals rather than checking every frame. When do you think will you add it to the simulator and the recorder? If it's soon-ish, I would suggest leaving it as is and mentioning this caveat in the simulator and recorder docs.

Reply:
I agree that it would be nicer with signals but I would prefer to use for now the checking per frame as grabbing things is definitely an interaction that should work in the simulator.
But you could keep the code in a comment and note that this would be preferred to do as soon as it is supported by the simulator and recorder.

I tried to add it once to the code but produced too many bugs (like signals triggered twice) and so reverted. It definitely will need some time and lot's of testing before this can be switched in the current state.